nob.h

```diff
@@ -486,6 +486,8 @@ void nob_cmd_append_null(Nob_Cmd *cmd, ...)
     }
 
     va_end(args);
+    nob_da_append(cmd, NULL);
```

Comment:
It is dangerous to do the Dynamic Array append on any DA you receive from function arguments. We don't know how it was constructed. It could've been constructed like this

```c
const char *args[] = {"echo", "Hello"};
Nob_Cmd cmd = {
    .items =  args,
    .count = NOB_ARRAY_LEN(args),
    .capacity = NOB_ARRAY_LEN(args)
}
nob_cmd_run_async(cmd);
```

I feel like passing a pointer to memory on the stack may upset `realloc` a bit. I solved it like this db69faebaf50bf6f90d615b7cb04ee1b115694a3